wcfsetup/install/files/lib/system/form/builder/container/GuestFormContainer.class.php

```diff
@@ -0,0 +1,146 @@
+<?php
+namespace wcf\system\form\builder\container;
+use wcf\system\event\EventHandler;
+use wcf\system\form\builder\field\CaptchaFormField;
+use wcf\system\form\builder\field\EmailFormField;
+use wcf\system\form\builder\field\user\UsernameFormField;
+
+/**
+ * Represents the whole container with a username, email and captcha input for forms supporting content generated by
+ * guests.
+ * 
+ * Instead of having to manually set up each individual component, this form container allows to
+ * simply create an instance of this class, set some required data for some components, and the
+ * setup is complete.
+ * 
+ * @author	Florian Gail
+ * @copyright	2001-2019 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package	WoltLabSuite\Core\System\Form\Builder\Container
+ * @since	5.2
+ */
+class GuestFormContainer extends FormContainer {
```

Comment:
Instead of adding the `GuestFormContainer` all the time, we should consider, if a simple `isEnabled` (or `isAppended`) make more sense, to make the generation of the form less complicated and append only a method call like `isEnabled(WCF::getSession()->userID !== null)` (or something like that). 

//cc @Gravatronics